flash/core/integrations/icevision/adapter.py

```diff
@@ -110,6 +110,27 @@ def _wrap_collate_fn(collate_fn, samples):
             DataKeys.METADATA: metadata,
         }
 
+    def change_collate_fn_dataloader(self, data_loader):
+        # Starting PL 1.7.0 - changing attributes after the DataLoader is initialized - will not work
+        # So we manually set the collate_fn to the wrapped version, for now.
+        new_kwargs = getattr(data_loader, "__pl_saved_kwargs", None)
+        new_collate_fn = functools.partial(self._wrap_collate_fn, data_loader.collate_fn)
+        if new_kwargs:
+            new_kwargs["collate_fn"] = new_collate_fn
+            setattr(data_loader, "__pl_saved_kwargs", new_kwargs)
+        data_loader.collate_fn = new_collate_fn
+        return data_loader
+
+    def update_collate_fn_dataloader(self, new_collate_fn, data_loader):
+        # Starting PL 1.7.0 - changing attributes after the DataLoader is initialized - will not work
+        # So we manually update the collate_fn for the dataloader, for now.
+        new_kwargs = getattr(data_loader, "__pl_saved_kwargs", None)
+        if new_kwargs:
+            new_kwargs["collate_fn"] = new_collate_fn
+            setattr(data_loader, "__pl_saved_kwargs", new_kwargs)
+        data_loader.collate_fn = new_collate_fn
+        return data_loader
```

Comment:
Seems like a lot of duplication. Seems like `change_collate_fn_dataloader` is equivalent to:
```
self.update_collate_fn_dataloader(functools.partial(self._wrap_collate_fn, data_loader.collate_fn), data_loader)
```

Maybe better to just do that and remove `change_collate_fn_dataloader`

Reply:
@ethanwharris, you are right, this could be merged into only one function. Btw, while technically not a *bug* on PL side, it is definitely a regression on user experience regarding dataloaders and I'm sorry @krshrimali had to run into that. I suspect though, he will not be the last one, so I will send a PR today/early next week to PL to improve UX in there

Reply:
@ethanwharris - Agreed. Fixed!

@otaj - Thank you for your help with this, please don't be sorry - looking forward to your PR which improves the UX further. Thank you again!